src/Hint/Bracket.hs

```diff
@@ -190,10 +190,14 @@ bracket pretty isPartialAtom root = f Nothing
     -- 'x' actually need bracketing in this context?
     f (Just (i, o, gen)) v@(remParens' -> Just x)
       | not $ needBracket i o x, not $ isPartialAtom (Just o) x =
-          rawIdea Suggestion msg (getLoc v) (pretty o) (Just (pretty (gen x))) [] [r] : g x
+          rawIdea Suggestion msg (getLoc v) from (Just to) [] [r] : g x
       where
         typ = findType v
         r = Replace typ (toSS v) [("x", toSS x)] "x"
+        (from, to) = reduceJunks (pretty o) (pretty (gen x))
+        reduceJunks :: String -> String -> (String, String)
```

Comment:
We don't need an intermediary function here.
And we have a clean `both` from `import Data.Tuple.Extra` for something like this:
```
(from, to) = both (lines . pretty) (o, gen x)
(from', to') = both unlines $ unzip $ filter (uncurry (/=)) (zip from to)
```